python/cog/server/http.py

```diff
@@ -113,6 +112,98 @@ def create_app(  # pylint: disable=too-many-arguments,too-many-locals,too-many-s
         # version=None # TODO
     )
 
+    # Pydantic 2 changes how optional fields are represented in OpenAPI schema.
+    # See: https://github.com/tiangolo/fastapi/pull/9873#issuecomment-1997105091
+    if PYDANTIC_V2:
+        from fastapi.openapi.utils import get_openapi
+
+        def remove_nullable_anyof(
+            openapi_schema: Union[Dict[str, Any], List[Dict[str, Any]]],
+        ) -> None:
+            if isinstance(openapi_schema, dict):
+                for key, value in list(openapi_schema.items()):
+                    if key == "anyOf" and isinstance(value, list):
+                        non_null_types = [
+                            item for item in value if item.get("type") != "null"
+                        ]
+                        if len(value) > len(non_null_types) == 1:
+                            openapi_schema.update(non_null_types[0])
+                            del openapi_schema[key]
+
+                            # FIXME: Update tests to expect nullable
+                            # openapi_schema["nullable"] = True
+
+                    else:
+                        remove_nullable_anyof(value)
+            elif isinstance(openapi_schema, list):  # pyright: ignore
+                for item in openapi_schema:
+                    remove_nullable_anyof(item)
+
+        def flatten_selected_allof_refs(
+            openapi_schema: Dict[str, Any],
+        ) -> None:
+            try:
+                response = openapi_schema["components"]["schemas"]["PredictionResponse"]
+                response["properties"]["output"] = {
+                    "$ref": "#/components/schemas/Output"
+                }
+            except KeyError:
+                pass
```

Comment:
Would this be expected to happen if their was no prediction response? And if so is that valid according to our schema?

Reply:
It shouldn't and wouldn't, no. In practice, I don't think this exception can really happen. But if it _were_ missing, raising a stink here seems unhelpful. The alternative would be to do safe dict navigation with conditional `.get` chaining, but that felt cumbersome.